beacon-chain/db/kv/attestations_test.go

```diff
@@ -56,6 +57,69 @@ func TestStore_AttestationCRUD(t *testing.T) {
 	}
 }
 
+func TestStore_BoltDontPanic(t *testing.T) {
+	db := setupDB(t)
+	defer teardownDB(t, db)
+	var wg sync.WaitGroup
+
+	for i := 0; i <= 100; i++ {
+		att := &ethpb.Attestation{
+			Data: &ethpb.AttestationData{
+				Crosslink: &ethpb.Crosslink{
+					Shard:      5,
+					ParentRoot: []byte("parent"),
+					StartEpoch: uint64(i + 1),
+					EndEpoch:   2,
+				},
+			},
+		}
+		ctx := context.Background()
+		attDataRoot, err := ssz.HashTreeRoot(att.Data)
+		if err != nil {
+			t.Fatal(err)
+		}
+		retrievedAtt, err := db.Attestation(ctx, attDataRoot)
+		if err != nil {
+			t.Fatal(err)
+		}
+		if retrievedAtt != nil {
+			t.Errorf("Expected nil attestation, received %v", retrievedAtt)
+		}
+		if err := db.SaveAttestation(ctx, att); err != nil {
+			t.Fatal(err)
+		}
+	}
+	// if indices are improperly deleted this test will then panic.
+	for i := 0; i <= 100; i++ {
+		wg.Add(1)
+		go func() {
+			att := &ethpb.Attestation{
+				Data: &ethpb.AttestationData{
+					Crosslink: &ethpb.Crosslink{
+						Shard:      5,
+						ParentRoot: []byte("parent"),
+						StartEpoch: uint64(i + 1),
+						EndEpoch:   2,
```

Comment:
```suggestion
						EndEpoch:   uint64(i + 2),
```
maybe at a later stage we will check that end is greater then start and this test will fail

Reply:
Not sure I understand, why will the test fail ? We just store and delete attestations from the db in the test

Reply:
i think you are right